rust/parquet/src/schema/types.rs

```diff
@@ -94,15 +94,38 @@ impl Type {
     /// Note that this will panic if called on a non-primitive type.
     pub fn get_physical_type(&self) -> PhysicalType {
         match *self {
-            Type::PrimitiveType {
-                basic_info: _,
-                physical_type,
-                ..
-            } => physical_type,
+            Type::PrimitiveType { physical_type, .. } => physical_type,
             _ => panic!("Cannot call get_physical_type() on a non-primitive type"),
         }
     }
 
+    /// Gets the type length of this primitive type.
+    /// Note that this will panic if called on a non-primitive type.
+    pub fn get_type_length(&self) -> i32 {
```

Comment:
These are used a bunch of times by `src/record/types/value.rs`; I needed functions to avoid duplicating this logic many times and decided this was the best place for them. I could move them there, or make these `pub(crate)`? Thoughts?

Reply:
I didn't mean to move these into a different place. Rather, may be you can put these few changes in a different (and much smaller) PR and have it reviewed/committed, then rebase this PR to use them?